src/main/java/com/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+package com.onlinebookstore.service.impl;
+
+import com.onlinebookstore.dto.order.CreateOrderRequestDto;
+import com.onlinebookstore.dto.order.OrderDto;
+import com.onlinebookstore.dto.order.OrderItemDto;
+import com.onlinebookstore.dto.order.OrderUpdateDto;
+import com.onlinebookstore.enums.Status;
+import com.onlinebookstore.exception.EntityNotFoundException;
+import com.onlinebookstore.mapper.OrderItemMapper;
+import com.onlinebookstore.mapper.OrderMapper;
+import com.onlinebookstore.model.CartItem;
+import com.onlinebookstore.model.Order;
+import com.onlinebookstore.model.OrderItem;
+import com.onlinebookstore.model.ShoppingCart;
+import com.onlinebookstore.model.User;
+import com.onlinebookstore.repository.cart.ShoppingCartRepository;
+import com.onlinebookstore.repository.order.OrderItemRepository;
+import com.onlinebookstore.repository.order.OrderRepository;
+import com.onlinebookstore.repository.user.UserRepository;
+import com.onlinebookstore.service.OrderService;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderDto createOrder(Long userId, CreateOrderRequestDto requestDto) {
+        User user = userRepository.findById(userId).orElseThrow(
+                () -> new EntityNotFoundException("Can't find user with id: " + userId));
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId).orElseThrow(
+                () -> new EntityNotFoundException("Can't find shopping cart for user with id: "
+                        + userId));
+        if (shoppingCart.getCartItems().isEmpty()) {
+            throw new EntityNotFoundException("Shopping cart is empty for user with id: " + userId);
```

Comment:
better create another exception. For example OrderProcessingException (it should return status CONFLICT)